tests/lint/unittest_lint.py

```diff
@@ -525,6 +525,158 @@ def test_load_plugin_command_line() -> None:
+                abspath="Command line or configuration file",
+                path="Command line or configuration file",
+                module="Command line or configuration file",
+                obj="",
+                line=1,
+                column=0,
+                end_line=None,
+                end_column=None,
+            ),
+        )
+
+        sys.path.remove(home_path)
+
+
+@pytest.mark.usefixtures("pop_pylintrc")
+def test_load_plugin_command_line_with_inithook_in_config_bad_value() -> None:
+    dummy_plugin_path = abspath(join(REGRTEST_DATA_DIR, "dummy_plugin"))
+    with fake_home() as home_path:
+        # construct a basic rc file that just modifies the path
+        pylintrc_file = join(home_path, "pylintrc")
+        with open(pylintrc_file, "w", encoding="utf8") as out:
+            out.writelines(
+                [
+                    "[MASTER]\n",
+                    f"init-hook=\"import sys; sys.path.append(r'{home_path}')\"\n",
+                ]
+            )
+
+        copytree(dummy_plugin_path, join(home_path, "copy_dummy"))
+
+        # To confirm we won't load this module _without_ the init hook running.
+        assert home_path not in sys.path
+
+        run = Run(
+            [
+                "--rcfile",
+                pylintrc_file,
+                "--load-plugins",
+                "copy_dummy",
+                join(REGRTEST_DATA_DIR, "empty.py"),
+            ],
+            reporter=testutils.GenericTestReporter(),
+            exit=False,
+        )
+        assert run._rcfile == pylintrc_file
+        assert home_path in sys.path
+        # The module should not be loaded
+        assert (
+            len(
+                [ch.name for ch in run.linter.get_checkers() if ch.name == "copy_dummy"]
+            )
+            == 0
+        )
+
+        # There should be a bad-plugin-message for this module
+        assert len(run.linter.reporter.messages) == 1
+        assert run.linter.reporter.messages[0] == Message(
+            msg_id="E0013",
+            symbol="bad-plugin-value",
+            msg="Plugin 'copy_dummy' is impossible to load, is it installed ? ('No module named 'copy_dummy'')",
+            confidence=interfaces.Confidence(
+                name="UNDEFINED",
+                description="Warning without any associated confidence level.",
+            ),
+            location=MessageLocationTuple(
+                abspath="Command line or configuration file",
+                path="Command line or configuration file",
+                module="Command line or configuration file",
+                obj="",
+                line=1,
+                column=0,
+                end_line=None,
+                end_column=None,
+            ),
+        )
+
+        sys.path.remove(home_path)
+
+
+def test_load_plugin_command_line_with_inithook_command_line() -> None:
+    regrtest_data_dir_abs = abspath(REGRTEST_DATA_DIR)
+
+    run = Run(
+        [
+            "--init-hook",
```

Comment:
Shall we add a test for the case where these are switched around? That shouldn't work and emit a warning.

Reply:
Actually, it would work fine — the CLI processing code has not been changed, and it just sort of _prepares_ the arguments, which are then handled in a fixed order later.

Reply:
Let's also test for that then 🎉 

I forgot I wrote that like that 😓

Reply:
😄 I'll add that test right now